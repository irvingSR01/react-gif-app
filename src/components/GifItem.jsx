import PropTypes from 'prop-types'

export const GifItem = ({ title, src, id }) => {
  return (
    <div className="card">
      <img src={src.url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

GifItem.PropTypes = {
  title: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired,
}