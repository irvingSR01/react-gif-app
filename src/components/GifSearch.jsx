import { useState } from "react"
import PropTypes from 'prop-types'

export const GifSearch = ({ onNewSearchValue }) => {

    const [search, setSearch] = useState('')

    const onInputChange = ({ target }) => {
        setSearch(target.value)
    }

    const onSubmitedForm = (e) => {
        e.preventDefault()

        if (search.trim().length <= 1) return;
        
        onNewSearchValue(search.trim())
        setSearch('')
    }
    
    return (
        <form onSubmit={onSubmitedForm} aria-label="form">
            <input type="text" value={search} onChange={onInputChange} />
        </form>
    )
}

GifSearch.propTypes = {
    onNewSearchValue: PropTypes.func.isRequired
}