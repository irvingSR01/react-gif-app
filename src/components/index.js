export * from './GifGrid'
export * from './GifItem'
export * from './GifSearch'
export * from './Title'