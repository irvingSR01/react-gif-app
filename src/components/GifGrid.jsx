import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from 'prop-types';

export const GifGrid = ({ currentSearchValue }) => {

  const { gifs, isLoading } = useFetchGifs( currentSearchValue );
  
  return (
    <>
    {
      isLoading && <h2>Cargando...</h2>
    }
      <div className="card-grid">
        {gifs.map((gif) => {
          return <GifItem key={gif.id} {...gif} />;
        })}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  currentSearchValue: PropTypes.string.isRequired,
}