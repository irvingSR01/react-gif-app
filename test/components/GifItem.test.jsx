import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en GifItem component', () => { 

    const title = 'One piece';
    const srcGif = { url: 'https://onepiece.com/onepiece.gif' };

    test('Debe hacer match con el snapshot', () => {

        const { container } = render( <GifItem title={title} src={srcGif} />)
        expect(container).toMatchSnapshot();
     })

     test('Debe mostrar la imagen con el url y alt indicados', () => {
        render( <GifItem title={title} src={srcGif} />)
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(src.url);
        // expect(screen.getByRole('img').alt).toBe(title);

        const { src, alt } = screen.getByRole('img')
        expect(src).toBe(srcGif.url);
        expect(alt).toBe(title);
      });

      test('Debe de mostrar el tÃ­tulo en el componente', () => {
        render (<GifItem title={title} src={srcGif.url} />);
        expect(screen.getByText(title)).toBeTruthy();
      })

 })