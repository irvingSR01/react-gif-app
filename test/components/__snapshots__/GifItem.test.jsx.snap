// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem component Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One piece"
      src="https://onepiece.com/onepiece.gif"
    />
    <p>
      One piece
    </p>
  </div>
</div>
`;
